src/resources/cpp/lambdas.md

```diff
@@ -0,0 +1,365 @@
+## Abstract
+
+A lambda expression is a shorthand notation for creating an unnamed callable object (also called a closure, or an
+anonymous function). A lambda can "capture" variables from its surrounding scope by value or by reference, allowing the
+body of the lambda to access or modify those variables without having to pass them as parameters. Unlike regular
+functions, lambdas are typically written in-line, combining the reusability of functions with direct access to local
+context. The lambda retains captured variables' state (for captures by value) or dynamically references them (for
+captures by reference), making lambdas ideal for short, context-dependent operations like custom comparisons, filters,
+or event handlers.
+
+For example:
```

Comment:
This example comes after a couple paragraphs on captures, but doesn't use any captures. Maybe it'd be good to show this code example early on, then talk about captures, then show an example using captures?